src/shadertools/lib/assemble-shaders.js

```diff
@@ -116,7 +116,18 @@ ${type === FRAGMENT_SHADER ? FRAGMENT_SHADER_PROLOGUE : ''}
   }
 
   // Add the actual source of this shader
-  assembledSource += source;
+  if (source.indexOf('#version ') === 0) {
```

Comment:
Nit: while this is functionally fine, to me the logic would be cleaner by separating this into two steps:
1) Early on split the `source` into `versionLine` & `coreSource`, where `versionLine` can be the empty string.
2) In the end, just append `versionLine` and `coreSource` in the appropriate positions, without any if statements.

Long term this would also allow us to deduce the version of the shader early on to do version dependent injection as well as do further breakdown of the core source to find additional injection points.